File: clients/java/examples/vectorize.java
// tag::pre-collection-definition[]
import com.datastax.astra.client.collections.Collection;
import com.datastax.astra.client.collections.definition.CollectionDefinition;
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.databases.Database;
import com.datastax.astra.client.collections.definition.documents.Document;
import com.datastax.astra.client.core.vector.SimilarityMetric;


public class Example {

  public static void main(String[] args) {
    // Instantiate the client
    DataAPIClient client = new DataAPIClient(new DataAPIClientOptions());

    // Connect to a database
    Database database =
        client.getDatabase(
            "API_ENDPOINT",
            new DatabaseOptions("APPLICATION_TOKEN", new DataAPIClientOptions()));
// end::pre-collection-definition[]

// tag::collection-definition-external-provider[]

    // Define the collection
    CollectionDefinition collectionDefinition =
        new CollectionDefinition()
            .vectorDimension(MODEL_DIMENSIONS)
            .vectorSimilarity(SimilarityMetric.SIMILARITY_METRIC)
            .vectorize(
                "{embedding-provider-name-api}",
                "MODEL_NAME",
                "API_KEY_NAME");
// end::collection-definition-external-provider[]

// tag::collection-definition-hugging-face-dedicated[]

    // Define parameters for the service provider
    Map<String, Object> parameters = new HashMap<>();
    parameters.put("endpointName", "ENDPOINT_NAME");
    parameters.put("regionName", "REGION");
    parameters.put("cloudName", "CLOUD_PROVIDER");

    // Define the collection
    CollectionDefinition collectionDefinition =
    new CollectionDefinition()
        .vectorDimension(MODEL_DIMENSIONS)
        .vectorSimilarity(SimilarityMetric.SIMILARITY_METRIC)
        .vectorize(
            "{embedding-provider-name-api}",
            "{embedding-provider-model-name-api}",
            "API_KEY_NAME",
            parameters);
// end::collection-definition-hugging-face-dedicated[]

// tag::collection-definition-openai[]

    // Define parameters for the service provider
    Map<String, Object> parameters = new HashMap<>();
    parameters.put("organizationId", "ORGANIZATION_ID");
    parameters.put("projectId", "PROJECT_ID");

    // Define the collection
    CollectionDefinition collectionDefinition =
    new CollectionDefinition()
        .vectorDimension(MODEL_DIMENSIONS)
        .vectorSimilarity(SimilarityMetric.SIMILARITY_METRIC)
        .vectorize(
            "{embedding-provider-name-api}",
            "MODEL_NAME",
            "API_KEY_NAME",
            parameters);
// end::collection-definition-openai[]

// tag::collection-definition-azure-openai[]

    // Define parameters for the service provider
    Map<String, Object> parameters = new HashMap<>();
    parameters.put("resourceName", "RESOURCE_NAME");
    parameters.put("deploymentId", "DEPLOYMENT_ID");

    // Define the collection
    CollectionDefinition collectionDefinition =
    new CollectionDefinition()
        .vectorDimension(MODEL_DIMENSIONS)
        .vectorSimilarity(SimilarityMetric.SIMILARITY_METRIC)
        .vectorize(
            "{embedding-provider-name-api}",
            "MODEL_NAME",
            "API_KEY_NAME",
            parameters);
// end::collection-definition-azure-openai[]

// tag::collection-definition-hosted-provider[]

    // Define the collection
    CollectionDefinition collectionDefinition =
        new CollectionDefinition()
            .vectorSimilarity(SimilarityMetric.COSINE)
            .vectorize(
                "{embedding-provider-name-api}",
                "{embedding-provider-model-name-api}");
// end::collection-definition-hosted-provider[]

// tag::post-collection-definition[]

    // Create the collection
    Collection<Document> collection = database.createCollection("COLLECTION_NAME", collectionDefinition);
  }
}
// end::post-collection-definition[]

File: clients/python/examples/vectorize.py
# tag::pre-collection-definition[]
from astrapy import DataAPIClient
from astrapy.constants import VectorMetric
from astrapy.info import (
    CollectionDefinition,
    CollectionVectorOptions,
    VectorServiceOptions,
)

# Instantiate the client
client = DataAPIClient()

# Connect to a database
database = client.get_database(
    "API_ENDPOINT",
    token="APPLICATION_TOKEN"
)
# end::pre-collection-definition[]

# tag::collection-definition-external-provider[]

# Define the collection
collection_definition = CollectionDefinition(
    vector=CollectionVectorOptions(
        metric=VectorMetric.SIMILARITY_METRIC,
        dimension=MODEL_DIMENSIONS,
        service=VectorServiceOptions(
            provider="{embedding-provider-name-api}",
            model_name="MODEL_NAME",
            authentication={
                "providerKey": "API_KEY_NAME",
            },
        )
    )
)
# end::collection-definition-external-provider[]

# tag::collection-definition-fluent-external-provider[]

# Define the collection
collection_definition = (
    CollectionDefinition.builder()
    .set_vector_dimension(MODEL_DIMENSIONS)
    .set_vector_metric(VectorMetric.SIMILARITY_METRIC)
    .set_vector_service(
        provider="{embedding-provider-name-api}",
        model_name="MODEL_NAME",
        authentication={
            "providerKey": "API_KEY_NAME",
        }
    )
    .build()
)
# end::collection-definition-fluent-external-provider[]

# tag::collection-definition-hugging-face-dedicated[]

# Define the collection
collection_definition = CollectionDefinition(
    vector=CollectionVectorOptions(
        metric=VectorMetric.SIMILARITY_METRIC,
        dimension=MODEL_DIMENSIONS,
        service=VectorServiceOptions(
            provider="{embedding-provider-name-api}",
            model_name="{embedding-provider-model-name-api}",
            authentication={
                "providerKey": "API_KEY_NAME",
            },
            parameters={
                "endpointName": "ENDPOINT_NAME",
                "regionName": "REGION",
                "cloudName": "CLOUD_PROVIDER",
            },
        )
    )
)
# end::collection-definition-hugging-face-dedicated[]

# tag::collection-definition-fluent-hugging-face-dedicated[]

# Define the collection
collection_definition = (
    CollectionDefinition.builder()
    .set_vector_dimension(MODEL_DIMENSIONS)
    .set_vector_metric(VectorMetric.SIMILARITY_METRIC)
    .set_vector_service(
        provider="{embedding-provider-name-api}",
        model_name="{embedding-provider-model-name-api}",
        authentication={
            "providerKey": "API_KEY_NAME",
        },
        parameters={
            "endpointName": "ENDPOINT_NAME",
            "regionName": "REGION",
            "cloudName": "CLOUD_PROVIDER",
        },
    )
    .build()
)
# end::collection-definition-fluent-hugging-face-dedicated[]

# tag::collection-definition-openai[]

# Define the collection
collection_definition = CollectionDefinition(
    vector=CollectionVectorOptions(
        metric=VectorMetric.SIMILARITY_METRIC,
        dimension=MODEL_DIMENSIONS,
        service=VectorServiceOptions(
            provider="{embedding-provider-name-api}",
            model_name="MODEL_NAME",
            authentication={
                "providerKey": "API_KEY_NAME",
            },
            parameters={
                "organizationId": "ORGANIZATION_ID",
                "projectId": "PROJECT_ID",
            },
        )
    )
)
# end::collection-definition-openai[]

# tag::collection-definition-fluent-openai[]

# Define the collection
collection_definition = (
    CollectionDefinition.builder()
    .set_vector_dimension(MODEL_DIMENSIONS)
    .set_vector_metric(VectorMetric.SIMILARITY_METRIC)
    .set_vector_service(
        provider="{embedding-provider-name-api}",
        model_name="MODEL_NAME",
        authentication={
            "providerKey": "API_KEY_NAME",
        },
        parameters={
            "organizationId": "ORGANIZATION_ID",
            "projectId": "PROJECT_ID",
        },
    )
    .build()
)
# end::collection-definition-fluent-openai[]

# tag::collection-definition-azure-openai[]

# Define the collection
collection_definition = CollectionDefinition(
    vector=CollectionVectorOptions(
        metric=VectorMetric.SIMILARITY_METRIC,
        dimension=MODEL_DIMENSIONS,
        service=VectorServiceOptions(
            provider="{embedding-provider-name-api}",
            model_name="MODEL_NAME",
            authentication={
                "providerKey": "API_KEY_NAME",
            },
            parameters={
                "resourceName": "RESOURCE_NAME",
                "deploymentId": "DEPLOYMENT_ID",
            },
        )
    )
)
# end::collection-definition-azure-openai[]

# tag::collection-definition-fluent-azure-openai[]

# Define the collection
collection_definition = (
    CollectionDefinition.builder()
    .set_vector_dimension(MODEL_DIMENSIONS)
    .set_vector_metric(VectorMetric.SIMILARITY_METRIC)
    .set_vector_service(
        provider="{embedding-provider-name-api}",
        model_name="MODEL_NAME",
        authentication={
            "providerKey": "API_KEY_NAME",
        },
        parameters={
            "resourceName": "RESOURCE_NAME",
            "deploymentId": "DEPLOYMENT_ID",
        },
    )
    .build()
)
# end::collection-definition-fluent-azure-openai[]

# tag::collection-definition-hosted-provider[]

# Define the collection
collection_definition = CollectionDefinition(
    vector=CollectionVectorOptions(
        metric=VectorMetric.COSINE,
        service=VectorServiceOptions(
            provider="{embedding-provider-name-api}",
            model_name="{embedding-provider-model-name-api}",
        )
    )
)
# end::collection-definition-hosted-provider[]

# tag::collection-definition-fluent-hosted-provider[]

# Define the collection
collection_definition = (
    CollectionDefinition.builder()
    .set_vector_metric(VectorMetric.COSINE)
    .set_vector_service(
        provider="{embedding-provider-name-api}",
        model_name="{embedding-provider-model-name-api}"
    )
    .build()
)
# end::collection-definition-fluent-hosted-provider[]

# tag::post-collection-definition[]

# Create the collection
collection = database.create_collection(
    "COLLECTION_NAME",
    definition=collection_definition,
)

print(f"* Collection: {collection.full_name}\n")
# end::post-collection-definition[]
